app/recipes/routes.py

```diff
@@ -0,0 +1,422 @@
+#!/usr/bin/env python3
+import html
+import math
+import re
+from app import mongo
+from bson.objectid import ObjectId
+from datetime import datetime
+from flask import Blueprint, current_app, render_template, redirect, request, url_for, flash, session, Markup
+from slugify import slugify
+
+
+# --------------------- #
+#    Flask Blueprint    #
+# --------------------- #
+recipes = Blueprint("recipes", __name__)
+
+
+# -------------------- #
+#    DB Collections    #
+# -------------------- #
+allergens_collection = mongo.db.allergens
+desserts_collection = mongo.db.desserts
+measurements_collection = mongo.db.measurements
+recipes_collection = mongo.db.recipes
+users_collection = mongo.db.users
+
+
+# ------------------- #
+#    Global Helper    #
+# ------------------- #
+@recipes.context_processor
+def desserts_total():
+    desserts_count = recipes_collection.count
+    return dict(desserts_count=desserts_count)
+
+
+# ---------------------- #
+#    Helper Functions    #
+# ---------------------- #
+
+# Allergens Dropdown List
+def dropdown_allergens():
+    return sorted([item for allergen in allergens_collection.find() for item in allergen.get("allergen_name")])
+
+
+# Dessert Type Dropdown List
+def dropdown_dessert_type():
+    return sorted([item for dessert in desserts_collection.find() for item in dessert.get("dessert_type")])
+
+
+# Measurements Dropdown List
+def dropdown_measurement():
+    return [item for measurement in measurements_collection.find() for item in measurement.get("measurement_unit")]
+
+
+# Find Recipe by ObjectID
+def get_recipe_id(recipe_id):
+    return recipes_collection.find_one({"_id": ObjectId(recipe_id)})
+
+
+# Find user's lowercase username
+def get_user_lower(user_param):
+    return users_collection.find_one({"username_lower": user_param.lower()})

```

Comment:
You should save the username on the database in the lower form as a norm... not having to use `username_lower`